src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat.slice(3);
+  const newSeparator = toFormat.slice(3);
```

Comment:
The `slice` method is being used on the `fromFormat` and `toFormat` arrays to obtain the separator. However, the `slice` method should be called with an index of `3` as the starting point and `4` as the ending point to extract only the separator element. Using `slice(3)` will return an array starting from the 4th element to the end, which is not the intended result. It should be corrected to `slice(3, 4)`.